src/daemon/emitter.cc

```diff
@@ -391,6 +406,7 @@ void Emitter::DoLocalExecute(const base::WorkerPool& pool) {
 void Emitter::DoRemoteExecute(const base::WorkerPool& pool, ResolveFn resolver,
                               const ui32 shard) {
   auto conf = this->conf();
+  std::random_device random_device;
```

Comment:
AFAICT `/dev/urandom` is the preferred option in this case, instead of `/dev/random` as latter may block, while urandom never blocks. Also `/dev/urandom` is a default source used in libc++.